custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py

```diff
@@ -149,6 +149,28 @@ def predict(self, url_params, form_params):
         self._stats_collector.disable()
         return response_status, response
 
+    @FlaskRoute(
+        "{}/predictUnstructured/".format(os.environ.get(URL_PREFIX_ENV_VAR_NAME, "")),
+        methods=["POST"],
+    )
+    def predict_unstructured(self, url_params, form_params):
+        if self._error_response:
+            return HTTP_513_DRUM_PIPELINE_ERROR, self._error_response
+
+        self._stats_collector.enable()
+        self._stats_collector.mark("start")
+
+        response, response_status = self.do_predict_unstructured()
```

Comment:
should be wrapped with try finally and mark(finish)+disable in finally block.

Reply:
ok